sdks/go/pkg/beam/core/runtime/graphx/serialize.go

```diff
@@ -318,6 +318,14 @@ func encodeType(t reflect.Type) (*v1.Type, error) {
 	if s, ok := tryEncodeSpecial(t); ok {
 		return &v1.Type{Kind: v1.Type_SPECIAL, Special: s}, nil
 	}
+	if k, ok := runtime.TypeKey(t); ok {
```

Comment:
<!--new_thread; commit:7f71ce6f0cadabb74e000c9fd67e8da53881472d; resolved:0-->
Interesting. So this gives users the power now to serialize something like a channel, as long as they register it and do something sane with it.

Reply:
The registered type still has to be a named type, but, yes, users can serialize the type of interfaces and whatnot using graphx. I think the old chan stuff was already there. However, we'd still need a coder registry that can code such types for them to appear in a user fn signature.